### DISCUSSION TITLE
Not sure if feature. what happens when etcdversion != etcdcluster

### DISCUSSION THREAD

**Discussion by fernacmk** [etcd-io/etcd#15503] [General]
I'm trying to have two apisix nodes running with etcd. I configure etcd in cluster mode among the two nodes. Apisix seems to want ETCD + 3.4.0, and I'm running 3.5.4

Now, if I start etcd with no parameters, I get for version the following string:
{"etcdserver":"3.5.4","etcdcluster":"3.5.0"}

However, If I start with etcd running in cluster mode, I get the following version string instead from a node:

{"etcdserver":"3.5.4","etcdcluster":"3.0.0"}

All nodes are running 3.5.4, same OS. All nodes have been cloned from a template & updated the same day.

Thing is, since in the cluster mode etcdcluster reports 3.0.0, apisix does not like that as they want 3.4.0+:

etcd cluster version 3.0.0 is less than the required version 3.4.0, please upgrade your etcd cluster

Any ideas? Is this working as expected?

**Comment by tjungblu** [comment]
this looks fairly odd. I've been just running the `gcr.io/etcd-development/etcd:v3.5.7` image on a local podman-compose cluster and I receive this from the /version endpoint:

> {"etcdserver":"3.5.7","etcdcluster":"3.5.0"}

Do you see anything interesting in your log? I for example see this:

> {"level":"info","ts":"2023-03-31T10:28:03.587Z","caller":"membership/cluster.go:576","msg":"updated cluster version","cluster-id":"8f2a2ec5c0087dcb","local-member-id":"d282ac2ce600c1ce","from":"3.0","to":"3.5"}

Can you share your etcd args and which image you're using?
